src/shared/components/pipeline-run-logs/logs/LogViewer.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import { Alert, Button, Toolbar, ToolbarContent, ToolbarItem } from '@patternfly/react-core';
+import { CompressIcon, DownloadIcon, ExpandIcon } from '@patternfly/react-icons/dist/esm/icons';
+import {
+  LogViewer as PatternFlyLogViewer,
+  LogViewerProps,
+  LogViewerSearch,
+} from '@patternfly/react-log-viewer';
+import classNames from 'classnames';
+import { saveAs } from 'file-saver';
+import { TaskRunKind } from '../../../../types';
+import { useFullscreen } from '../../../hooks/fullscreen';
+import { LoadingInline } from '../../status-box/StatusBox';
+import LogsTaskDuration from './LogsTaskDuration';
+
+export type Props = LogViewerProps & {
+  showSearch?: boolean;
+  data: string;
+  autoScroll?: boolean;
+  downloadAllLabel?: string;
+  onDownloadAll?: () => Promise<Error>;
+  taskRun?: TaskRunKind;
+  isLoading: boolean;
+  errorMessage: string | null;
+};
+
+const LogViewer: React.FC<Props> = ({
+  showSearch = true,
+  autoScroll,
+  data = '',
+  downloadAllLabel,
+  onDownloadAll,
+  taskRun,
+  isLoading,
+  errorMessage,
+  ...props
+}) => {
+  const taskName = taskRun?.spec.taskRef?.name ?? taskRun?.metadata.name;
+  const scrolledRow = React.useMemo(
+    () => (autoScroll ? data.split('\n').length : 0),
+    [autoScroll, data],
+  );
+
+  const [isFullscreen, fullscreenRef, fullscreenToggle] = useFullscreen<HTMLDivElement>();
+  const [downloadAllStatus, setDownloadAllStatus] = React.useState(false);
+
+  const downloadLogs = () => {
+    if (!data) return;
+    const blob = new Blob([data], {
+      type: 'text/plain;charset=utf-8',
+    });
+    saveAs(blob, `${taskName}.log`);
+  };
+
+  const startDownloadAll = () => {
+    setDownloadAllStatus(true);
+    onDownloadAll()
+      .then(() => {
+        setDownloadAllStatus(false);
+      })
+      .catch((err: Error) => {
+        setDownloadAllStatus(false);
+        // eslint-disable-next-line no-console
+        console.warn(err.message || 'Error downloading logs.');
+      });
+  };
+  return (
+    <div ref={fullscreenRef} style={{ height: isFullscreen ? '100vh' : '100%' }}>
+      <PatternFlyLogViewer
+        {...props}
+        hasLineNumbers={false}
+        height={isFullscreen ? '100%' : undefined}
+        data={data}
+        theme="dark"
```

Comment:
Should we allow use to switch between the light/dark theme? WDYT?

Reply:
that's a good idea, but I would not use the value from `useTheme` directly, since it would cause some awkwardness from the users that use light theme and start seeing the Logs in a light theme as default. I'd suggest to add a "theme switcher" in the Logs' header that would use "dark" as default. And the user would be able to change the Logs theme apart from the "main application theme" - WDYT?